core/blockchain_reader.go

```diff
@@ -230,6 +230,20 @@ func (bc *BlockChain) GetReceiptsByHash(hash common.Hash) types.Receipts {
 	return receipts
 }
 
+// GetRawReceiptsByHash retrieves the receipts for all transactions in a given block
+// without deriving the internal fields and the Bloom.
+func (bc *BlockChain) GetRawReceiptsByHash(hash common.Hash) types.Receipts {
```

Comment:
This is probably the wrong thing. The advantage of the new receipt "type" is that it's essentially the `storageReceipt`, i.e. the form we already have it in the database. So you could just return the RLP without even decoding it in the first place. Would save a lot of CPU cycles.

Reply:
And also, probably we can define a `rawReceiptCache` in blockchain

Reply:
The receiptsCache is only 32 items in size (which is probably correct for getReceiptsByHash), but for historical receipts, we will probably never use it. I don't think we are using the `getReceipts` function for the head block